hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/sdk/TestAWSV2SDK.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.fs.sdk;
+import org.junit.Test;
+import org.apache.hadoop.test.AbstractHadoopTestBase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * Tests to verify AWS SDK based issues like duplicated shaded classes and others
+ */
+public class TestAWSV2SDK extends AbstractHadoopTestBase {
+
+    Logger LOG = LoggerFactory.getLogger(this.getClass().getName());
+    @Test
+    public void testShadedClasses() throws IOException {
+        String allClassPath = System.getProperty("java.class.path");
+        LOG.debug("Current classpath:{}", allClassPath);
+        String[] classPaths = allClassPath.split(File.pathSeparator);
+        String v2ClassPath = null;
+        for(String classPath : classPaths){
+            //Checking for only version 2.x sdk here
+            if (classPath.contains("awssdk/bundle/2")) {
+                v2ClassPath = classPath;
+                break;
+            }
+        }
+        assertNotEquals("AWS V2 SDK should be present on the classpath",
+                v2ClassPath, null);
+        List<String> listOfV2SdkClasses = getClassNamesFromJarFile(v2ClassPath);
+        String awsSdkPrefix = "software/amazon/awssdk";
+        List<String> unshadedClasses = new ArrayList<>();
+        for(String awsSdkClass : listOfV2SdkClasses){
+            if (!awsSdkClass.startsWith(awsSdkPrefix)) {
```

Comment:
hadoop uses two space indentation

Reply:
Can I have the xml for the intellij settings, I think my new mac doesn't have all the settings?